lib/node-conversions.js

```diff
@@ -62,23 +62,23 @@ if (NodeReadableStream) {
     async _read(size) {
       try {
         while (true) {
-          const { done, value } = await this._reader.read()
+          const { done, value } = await this._reader.read();
           if (done) {
             this.push(null);
             break;
           }
-          if (!this.push(value) || this._cancelling) {
-            this._reading = false;
+          if (!this.push(value)) {
             break;
           }
         }
-      } catch(e) {
-        this.emit('error', e);
+      } catch (e) {
+        this.destroy(e);
       }
     }
 
-    _destroy(reason) {
-      this._reader.cancel(reason);
+    _destroy(error, callback) {
+      this._reader.cancel(error);
+      callback(error);
     }
```

Comment:
Since `reader.cancel` is actually async, I assume this should be
```suggestion
    async _destroy(error, callback) {
      await this._reader.cancel(error);
      callback(error);
    }
```
, right?

Also, are you sure that `reason` should be passed to the callback? Isn't it just in case cancelling the stream failed? Cancelling a Web Stream always succeeds, so I think there should be no error passed?

Reply:
> Cancelling a Web Stream always succeeds

Promise is promise we ought to expect it can throw. There is no guarantee Web Stream implementation wouldn't change in the future. We may also deal with some kind of ponyfill that implement Web Stream in an unexpected way.

Reply:
OK. In that case, how about simply:
```js
    _destroy(error, callback) {
      this._reader.cancel(error).then(callback, callback);
    }
```
Since `this._reader.cancel(error)` resolves with `error`, this will pass it along to `callback`, or in case it rejects the promise, it'll call the callback with that error.

Reply:
> how about simply

`await` is superior in a way that it supports promise-returning functions and non-promise-returning functions alike.

<img width="544" alt="image" src="https://user-images.githubusercontent.com/75836966/212296122-ef3220a9-41a1-4588-b1dd-538eeb2d2c40.png">

I believe the simple implementation would result in a bug when it meets the stub:
https://github.com/openpgpjs/web-stream-tools/blob/f301931e3e0746992ed7e0a32683ec05e377bbd5/lib/reader.js#L23

Reply:
Apologies for the delay.

> I believe the simple implementation would result in a bug when it meets the stub:

I think we should probably fix that, then. The ArrayStream stubs are anyway a bit hacky, and indeed incomplete. I'll take a stab at fixing them, unless you feel like doing so?